graphistry/ArrowFileUploader.py

```diff
@@ -109,72 +109,130 @@ def post_arrow(self, arr: pa.Table, file_id: str, url_opts: str = 'erase=true')
 
         try:
             out = res.json()
-            logger.debug('Server upload file response: %s', out)
-            if not out['is_valid']:
-                if out['is_uploaded']:
-                    raise RuntimeError("Uploaded file contents but cannot parse (file_id still valid), see errors", out['errors'])
+            logger.debug("Server upload file response: %s", out)
+            if not out["is_valid"]:
+                if out["is_uploaded"]:
+                    raise RuntimeError(
+                        "Uploaded file contents but cannot parse "
+                        "(file_id still valid), see errors",
+                        out["errors"],
+                    )
                 else:
-                    raise RuntimeError("Erased uploaded file contents upon failure (file_id still valid), see errors", out['errors'])
-            return out
+                    raise RuntimeError(
+                        "Erased uploaded file contents upon failure "
+                        "(file_id still valid), see errors",
+                        out["errors"],
+                    )
         except Exception as e:
-            logger.error('Failed uploading file: %s', res.text, exc_info=True)
+            text = res.text if res.text else str(res)
+            logger.error("Failed uploading file: %s", text, exc_info=True)
             raise e
 
     ###
 
     def create_and_post_file(
-        self, arr: pa.Table, file_id: Optional[str] = None, file_opts: dict = {}, upload_url_opts: str = 'erase=true', memoize: bool = True
-    ) -> Tuple[str, dict]:
```

Comment:
changing the return type here may be a breaking change, should we preserve the old return type, or do a more subtle breaking change like `Tuple[str, Optional[dict]]` , where first call returns the response, and subsequent return None?

I see similar elsewhere, like `post_arrow` / `_post_arrow`... these are public methods that breaking changes don't seem too motivated on wrt this pr

Reply:
reverted, I thought of this as internal